Input.cpp & test fail with clang 3.5 - reference cannot be bound to dereferenced null pointer

env: c++11, clang-3.5, using stdlibc++ (gcc4.8)
the the line in Input.cpp: 
https://github.com/numenta/nupic.core/blob/master/src/main/engine/Input.cpp#L44

causes a warning: 
`[ 58%] Building CXX object CMakeFiles/nupic_core.dir/main/engine/Network.cpp.o                                                                                                                                                   
/home/mmm/nupic/nupic.core/src/main/engine/Input.cpp:44:8: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; comparison may be assumed to always evaluate to false
      [-Wtautological-undefined-compare]
  if (&region == NULL)
       ^~~~~~    ~~~~
`
which in turn produces an error in tests. 
- Total failed   = 1
  FAILED: InputTest
- Total critical failures = 0
- Success rate = 99.9542

`======= Tests for InputTest ==============
Total tests: 88, Failures:      1, Disabled     : 0
FAIL  statement 'Input z(*rn, NTA_BasicType_Int32, true)' should fail (line: 47)
      Expected result: 1
      Actual result:   0
`

https://github.com/numenta/nupic.core/blob/master/src/test/testeverything/engine/InputTest.cpp#L47

--- a/src/main/engine/Input.cpp
+++ b/src/main/engine/Input.cpp
@@ -41,10 +41,6 @@ Input::Input(Region& region, NTA_BasicType dataType, bool isRegionLevel) :
   region_(region), isRegionLevel_(isRegionLevel), 
   initialized_(false),  data_(dataType), name_("Unnamed")
 {
-  if (&region == NULL)
-  {
-    NTA_THROW << "Attempt to create Input with a null region";
-  }
 }
 
 Input::~Input()

--- a/src/test/testeverything/engine/InputTest.cpp
+++ b/src/test/testeverything/engine/InputTest.cpp
@@ -43,8 +43,6 @@ void InputTest::RunTests()
     //Test constructor
     Input x(*r1, NTA_BasicType_Int32, true);
     Input y(*r2, NTA_BasicType_Byte, false);
-    Region * rn = NULL;
-    SHOULDFAIL(Input z(*rn, NTA_BasicType_Int32, true));
     SHOULDFAIL(Input i(*r1, (NTA_BasicType)(NTA_BasicType_Last + 1), true));
 
     //test getRegion()

